modules/3d/test/test_pose_graph.cpp

```diff
@@ -155,6 +155,109 @@ TEST(PoseGraph, sphereG2O)
+
+    // The dataset was taken from here: https://lucacarlone.mit.edu/datasets/
+    // Connected paper:
+    // L.Carlone, R.Tron, K.Daniilidis, and F.Dellaert.
+    // Initialization Techniques for 3D SLAM : a Survey on Rotation Estimation and its Use in Pose Graph Optimization.
+    // In IEEE Intl.Conf.on Robotics and Automation(ICRA), pages 4597 - 4604, 2015.
+
+    std::string filename = cvtest::TS::ptr()->get_data_path() + "/cv/rgbd/sphere_bignoise_vertex3.g2o";
+
+    Ptr<detail::PoseGraph> pgOptimizerOnly = readG2OFile(filename);
+    Ptr<detail::PoseGraph> pgWithMSTAndOptimizer = readG2OFile(filename);
+    Ptr<detail::PoseGraph> init = readG2OFile(filename);
+
+    pgWithMSTAndOptimizer->initializePosesWithMST();
+
+    // You may change logging level to view detailed optimization report
+    // For example, set env. variable like this: OPENCV_LOG_LEVEL=INFO
+
+    // geoScale=1 is experimental, not guaranteed to work on other problems
+    // the rest are default params
+    pgOptimizerOnly->createOptimizer(LevMarq::Settings().setGeoScale(1.0)
+                        .setMaxIterations(100)
+                        .setCheckRelEnergyChange(true)
+                        .setRelEnergyDeltaTolerance(1e-6)
+                        .setGeodesic(true));
+    pgWithMSTAndOptimizer->createOptimizer(LevMarq::Settings().setGeoScale(1.0)
+                        .setMaxIterations(100)
+                        .setCheckRelEnergyChange(true)
+                        .setRelEnergyDeltaTolerance(1e-6)
+                        .setGeodesic(true));
+
+    auto r1 = pgWithMSTAndOptimizer->optimize();
+    auto r2 = pgOptimizerOnly->optimize();
+
+    EXPECT_TRUE(r1.found);
+    EXPECT_TRUE(r2.found);
+    EXPECT_LE(r2.energy, 1.47723e+06);
+    // Allow small tolerance due to optimization differences; final energy/iterations are effectively the same
+    EXPECT_LE(std::abs(r1.energy - r2.energy), 1e-2);
+    ASSERT_LE(std::abs(r1.iters - r2.iters), 1);
+
+    // Add the "--test_debug" to arguments to see resulting pose graph nodes positions
+    if (cvtest::debugLevel > 0)
+    {
+        // Write OBJ for MST-initialized pose graph with optimizer
+        {
+            std::string fname = "pg_with_mst_and_optimizer.obj";
+            std::fstream of(fname, std::fstream::out);
+            std::vector<size_t> ids = pgWithMSTAndOptimizer->getNodesIds();
+            for (const size_t& id : ids)
+            {
+                Point3d d = pgWithMSTAndOptimizer->getNodePose(id).translation();
+                of << "v " << d.x << " " << d.y << " " << d.z << std::endl;
+            }
+            size_t esz = pgWithMSTAndOptimizer->getNumEdges();
+            for (size_t i = 0; i < esz; i++)
+            {
+                size_t sid = pgWithMSTAndOptimizer->getEdgeStart(i), tid = pgWithMSTAndOptimizer->getEdgeEnd(i);
+                of << "l " << sid + 1 << " " << tid + 1 << std::endl;
+            }
+            of.close();
+        }
```

Comment:
We have saveMesh / savePointCloud for obj and ply i/o. Please use it: https://docs.opencv.org/5.x/da/d35/group____3d.html#gad565f2878b3d8c3f959ae7967a520bb2